tests/test_input_damages.py

```diff
@@ -0,0 +1,1287 @@
+        )
+
+        xr.testing.assert_equal(ds_out_expected, ds_out_actual)
+
+
+def test_compute_ag_damages(
+    tmp_path,
+    econvars_fixture,
+):
+    """
+    Test that compute_ag_damages correctly reshapes ag estimate runs for use in integration system
+    """
+    rcp = ["rcp45", "rcp85"]
+    gcm = ["ACCESS1-0", "GFDL-CM3"]
+    model = ["low", "high"]
+    ssp = ["SSP2", "SSP3"]
+    batch = ["batch3", "batch6", "batch9"]
+
+    for r in rcp:
+        for g in gcm:
+            for m in model:
+                for s in ssp:
+                    for b in batch:
+                        d = os.path.join(tmp_path, "ag_in", b, r, g, m, s)
+                        if not os.path.exists(d):
+                            os.makedirs(d)
+                        infile = os.path.join(d, "disaggregated_damages.nc4")
+
+                        ds_in = xr.Dataset(
+                            {
+                                "wc_no_reallocation": (
+                                    [
+                                        "gcm",
+                                        "model",
+                                        "rcp",
+                                        "ssp",
+                                        "batch",
+                                        "region",
+                                        "year",
+                                        "variable",
+                                        "demand_topcode",
+                                        "Es_Ed",
+                                    ],
+                                    np.full((1, 1, 1, 1, 1, 2, 2, 1, 1, 1), 2),
+                                ),
+                            },
+                            coords={
+                                "region": (
+                                    ["region"],
+                                    ["ZWE.test_region", "USA.test_region"],
+                                ),
+                                "year": (["year"], [2010, 2099]),
+                                "ssp": (["ssp"], [s]),
+                                "iam": (["iam"], [m]),
+                                "continent": (["region"], ["Africa", "Americas"]),
+                                "variable": (["variable"], ["gdp"]),
+                                "demand_topcode": (["demand_topcode"], ["agshare_10"]),
+                                "Es_Ed": (["Es_Ed"], ["0.1_-0.04"]),
+                                "gcm": (["gcm"], [g]),
+                                "rcp": (["rcp"], [r]),
+                                "batch": (["batch"], [b]),
+                                "model": (["model"], [m]),
+                                "market_level": (["market_level"], ["continent"]),
+                            },
+                        )
+
+                        ds_in_save = ds_in.sel(iam=m, market_level="continent")
+
+                        ds_in_save.to_netcdf(infile)
```

Comment:
I'm not seeing where these files are used. Are these automatically read in in the `compute_ag_damages` func? Is the path hard-coded? I would have thought `input_path` argument would set this but the path looks different.

Reply:
Yes, they are used by `compute_ag_damages` func, and the path is not hard-coded actually. Input Ag impacts created for tests are saved under directories named in the convention `/tmp_path/ag_in/batch/rcp/gcm/iam/ssp`:
https://github.com/ClimateImpactLab/dscim/blob/1aaa8e286ad7e269c5110ee46634e0a98cced6c7/tests/test_input_damages.py#L379-L382

The `input_path` argument for `compute_ag_damages` func is set to be the root directory of all input files (which is `/tmp_path/ag_in`):
https://github.com/ClimateImpactLab/dscim/blob/1aaa8e286ad7e269c5110ee46634e0a98cced6c7/tests/test_input_damages.py#L444-L445

and `compute_ag_damages` func knows how to retrieve input structure to read files:
https://github.com/ClimateImpactLab/dscim/blob/1aaa8e286ad7e269c5110ee46634e0a98cced6c7/src/dscim/preprocessing/input_damages.py#L315

Reply:
I see, that's helpful. Thanks!